go/vt/vtctl/workflow/materializer.go

```diff
@@ -441,10 +438,17 @@ func (mz *materializer) buildMaterializer() error {
 	if err != nil {
 		return err
 	}
-	if len(ms.SourceShards) > 0 {
+	var specifiedTargetShards []string
+	switch {
+	case mz.IsMultiTenantMigration():
+		specifiedTargetShards = ms.WorkflowOptions.Shards
+	case len(ms.SourceShards) > 0: // shard-by-shard migration
+		specifiedTargetShards = ms.SourceShards
+	}
+	if len(specifiedTargetShards) > 0 {
 		var targetShards2 []*topo.ShardInfo
```

Comment:
A comment here and on line 422 will be helpful. We are choosing only valid serving shards from the ones provided by the workflow options aka the user.